testing/kokoro/system_tests.sh

```diff
@@ -97,75 +194,21 @@ if [[ $KOKORO_BUILD_ARTIFACTS_SUBDIR = *"system-tests"* ]]; then
   ./testing/kokoro/configure_gcloud.bash;
 fi
 
-RUN_ALL_TESTS="0"
-# If this is a nightly test (not a PR), run all tests.
-if [ -z ${KOKORO_GITHUB_PULL_REQUEST_NUMBER:-} ]; then
-  RUN_ALL_TESTS="1"
-fi
-
-# Also see trampoline.sh - system_tests.sh is only run for PRs when there are
-# significant changes.
-SIGNIFICANT_CHANGES=$(git --no-pager diff --name-only HEAD..master | egrep -v '(\.md$|^\.github)' || true)
-# CHANGED_DIRS is the list of significant top-level directories that changed.
-# CHANGED_DIRS will be empty when run on master.
-CHANGED_DIRS=$(echo $SIGNIFICANT_CHANGES | tr ' ' '\n' | grep "/" | cut -d/ -f1 | sort -u | tr '\n' ' ')
-
-if echo $SIGNIFICANT_CHANGES | tr ' ' '\n' | grep "^go.mod$" || [[ $CHANGED_DIRS =~ "testing" || $CHANGED_DIRS =~ "internal" ]]; then
-  RUN_ALL_TESTS="1"
-fi
-
-# Filter out directories that don't exist (the current PR deleted them).
-TARGET_DIRS=""
-for d in $CHANGED_DIRS; do
-  if [ -d "$d" ]; then
-    TARGET_DIRS="$TARGET_DIRS $d"
-  fi
-done
+date
 
 if [[ $RUN_ALL_TESTS = "1" ]]; then
-  TARGET="./..."
+  GO_TEST_TARGET="./..."
   echo "Running all tests"
 elif [[ -z "${TARGET_DIRS// }" ]]; then
-  TARGET=""
+  GO_TEST_TARGET=""
   echo "Only running root tests"
 else
-  TARGET=$(printf "./%s/... " $TARGET_DIRS)
+  GO_TEST_TARGET=$(printf "./%s/... " $TARGET_DIRS)
   echo "Running tests in modified directories: $TARGET"
 fi
 
-# Do the easy stuff before running tests. Fail fast!
-if [ $GOLANG_SAMPLES_GO_VET ]; then
-  diff -u <(echo -n) <(gofmt -d -s .)
-
-  # Generate a list of all go files not inside a go submodule.
-  files=$(find $TARGET_DIRS \( -exec [ -f {}/go.mod ] \; -prune \) -o -name "*.go" -print)
-  # Display the list of files to facilitate troubleshooting.
-  # Current risk behavior: if find outputs anything to stdout go vet is attempted.
-  echo "$files"
-
-  # If there are no go files, skip go vet $TARGET.
-  if [ -z "$files" ]; then
-    echo "No *.go files found, skipping go vet on $TARGET"
-  else
-    go vet $TARGET
-  fi
-
-  # Run go vet inside each sub-module.
-  # Recursive submodules are not supported.
-  for i in $(find "$TARGET_DIRS" -name 'go.mod')
-  do
-    mod="$(dirname $i)"
-    echo "Running 'go vet' in $mod"
-    pushd "$mod"
-    go vet ./...
-    popd
-  done
-fi
-
-date
-
 OUTFILE=gotest.out
-2>&1 go test -timeout $TIMEOUT -v . $TARGET | tee $OUTFILE
+2>&1 go test -timeout $TIMEOUT -v . $GO_TEST_TARGET | tee $OUTFILE
```

Comment:
Does this end up with a single output file for all of the tests in all of the modules?

Reply:
This change doesn't address submodule recursion for `go test`, my plan was to do that in a follow-up but I can add that to this PR.

Reply:
I have now added test support. Please look again.